web/app/js/components/BreadcrumbHeader.jsx

```diff
@@ -0,0 +1,109 @@
+import _ from 'lodash';
+import PropTypes from "prop-types";
+import React from 'react';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { withContext } from './util/AppContext.jsx';
+import { Breadcrumb, Layout } from 'antd';
+import { friendlyTitle, isResource, singularResource } from "./util/Utils.js";
+import './../../css/breadcrumb-header.css';
+
+class BreadcrumbHeader extends React.Component {
+
+  static propTypes = {
+    api: PropTypes.shape({
+      PrefixedLink: PropTypes.func.isRequired,
+    }).isRequired,
+    location: ReactRouterPropTypes.location.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.api = this.props.api;
+  }
+
+  processResourceDetailURL(segments) {
+    if (segments.length === 4) {
+      let splitSegments = _.chunk(segments, 2);
+      let resourceNameSegment = splitSegments[1];
+      resourceNameSegment[0] = singularResource(resourceNameSegment[0]);
+      return _.concat(splitSegments[0], resourceNameSegment.join('/'));
+    } else {
+      return segments;
+    }
+  }
+
+  convertURLToBreadcrumbs(location) {
+    if (location.length === 0) {
+      return [];
+    } else {
+      let segments = location.split('/').slice(1);
+      let finalSegments = this.processResourceDetailURL(segments);
+
+      return _.map(finalSegments, segment => {
+        let partialUrl = _.takeWhile(segments, s => {
+          return s !== segment;
+        });
+
+        if (partialUrl.length !== segments.length) {
+          partialUrl.push(segment);
+        }
+
+        return {
+          link: `/${partialUrl.join("/")}`,
+          segment: segment
+        };
+      });
+    }
+  }
+
+  renderBreadcrumbSegment(segment, index) {
```

Comment:
we have few enough routes that we can get away with a straight up map of routes to title we want to display rather than special casing a bunch of things in this way, e.g.
```
  const routeToCrumbTitle = {
    "servicemesh": "Service Mesh",
    "replicationcontroller": "Replication Controller"
  }

let titleToDisplay = routeToCrumbTitle[routeSegment] || _.startCase(routeSegment)
```